targets/emby/api.go

```diff
@@ -0,0 +1,168 @@
+package emby
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/cloudbox/autoscan"
+	"github.com/rs/zerolog"
+	"net/http"
+)
+
+type apiClient struct {
+	url   string
+	token string
+	log   zerolog.Logger
+}
+
+func newApiClient(c Config) *apiClient {
+	return &apiClient{
+		url:   c.URL,
+		token: c.Token,
+		log: autoscan.GetLogger(c.Verbosity).With().
+			Str("target", "emby").
+			Str("url", c.URL).
+			Logger(),
+	}
+}
+
+func (c apiClient) Available() error {
+	// create request
+	req, err := http.NewRequest("GET", autoscan.JoinURL(c.url, "emby", "System", "Info"), nil)
+	if err != nil {
+		return fmt.Errorf("failed creating availability request: %v: %w", err, autoscan.ErrFatal)
+	}
+
+	// set headers
+	req.Header.Set("X-Emby-Token", c.token)
+	req.Header.Set("Accept", "application/json")
+
+	// send request
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return fmt.Errorf("failed sending availability request: %v: %w", err, autoscan.ErrTargetUnavailable)
+	}
+
+	defer res.Body.Close()
+
+	// validate response
+	if res.StatusCode != 200 {
+		return fmt.Errorf("%v: failed validating availability request response: %w",
+			res.Status, autoscan.ErrTargetUnavailable)
+	}
+
+	return nil
+}
+
+type library struct {
+	ID   string
+	Name string
+	Path string
+}
+
+func (c apiClient) Libraries() ([]library, error) {
+	// create request
+	req, err := http.NewRequest("GET",
+		autoscan.JoinURL(c.url, "emby", "Library", "SelectableMediaFolders"), nil)
```

Comment:
Can this be refactored into two lines as seen in [plex.go](https://github.com/Cloudbox/autoscan/blob/e2c376cc8504fa8ad96ef9773d1f7b830ab2647f/targets/plex/scan.go#L84)?

```suggestion
	reqURL := autoscan.JoinURL(c.url, "emby", "Library", "SelectableMediaFolders")
	req, err := http.NewRequest("GET", reqURL, nil)
```